addons/sale/models/sale_order.py

```diff
@@ -1111,6 +1113,22 @@ def _notify_get_recipients_groups(self, msg_vals=None):
 
         return groups
 
+    def _notify_by_email_prepare_rendering_context(self, message, msg_vals, model_description=False,
+                                                   force_email_company=False, force_email_lang=False):
+        render_context = super()._notify_by_email_prepare_rendering_context(
+            message, msg_vals, model_description=model_description,
+            force_email_company=force_email_company, force_email_lang=force_email_lang
+        )
+        if self.validity_date:
+            amount_txt = _('%(amount)s due %(date)s',
+                           amount=format_amount(self.env, self.amount_total, self.pricelist_id.currency_id, lang_code=render_context.get('lang')),
```

Comment:
The "currency_id" looks like it's a related (to the pricelist) stored field, maybe better to use that?

Reply:
Updated